tests/test_scalars.py

```diff
@@ -24,17 +39,685 @@
+    check(assert_type(td.to_numpy(), np.timedelta64), np.timedelta64)
+    check(assert_type(td.to_pytimedelta(), dt.timedelta), dt.timedelta)
+    check(assert_type(td.to_timedelta64(), np.timedelta64), np.timedelta64)
+    check(assert_type(td.total_seconds(), float), float)
+    check(assert_type(td.view(np.int64), object), np.int64)
+    check(assert_type(td.view("i8"), object), np.int64)
+
+
+def test_timedelta_add_sub() -> None:
+    td = pd.Timedelta("1 day")
+
+    ndarray_td64: npt.NDArray[np.timedelta64] = np.array(
+        [1, 2, 3], dtype="timedelta64[D]"
+    )
+    ndarray_dt64: npt.NDArray[np.datetime64] = np.array(
+        [1, 2, 3], dtype="datetime64[D]"
+    )
+    as_period = pd.Period("2012-01-01", freq="D")
+    as_timestamp = pd.Timestamp("2012-01-01")
+    as_datetime = dt.datetime(2012, 1, 1)
+    as_date = dt.date(2012, 1, 1)
+    as_datetime64 = np.datetime64(1, "ns")
+    as_dt_timedelta = dt.timedelta(days=1)
+    as_timedelta64 = np.timedelta64(1, "D")
+    as_timedelta_index = pd.TimedeltaIndex([td])
+    as_timedelta_series = pd.Series(as_timedelta_index)
+    as_period_index = pd.period_range("2012-01-01", periods=3, freq="D")
+    as_datetime_index = pd.date_range("2012-01-01", periods=3)
+    as_ndarray_td64 = ndarray_td64
+    as_ndarray_dt64 = ndarray_dt64
+    as_nat = pd.NaT
+
+    check(assert_type(td + td, pd.Timedelta), pd.Timedelta)
+    check(assert_type(td + as_period, pd.Period), pd.Period)
+    check(assert_type(td + as_timestamp, pd.Timestamp), pd.Timestamp)
+    check(assert_type(td + as_datetime, pd.Timestamp), pd.Timestamp)
+    check(assert_type(td + as_date, dt.date), dt.date)
+    check(assert_type(td + as_datetime64, pd.Timestamp), pd.Timestamp)
+    check(assert_type(td + as_dt_timedelta, pd.Timedelta), pd.Timedelta)
+    check(assert_type(td + as_timedelta64, pd.Timedelta), pd.Timedelta)
+    check(assert_type(td + as_timedelta_index, pd.TimedeltaIndex), pd.TimedeltaIndex)
+    check(
+        assert_type(td + as_timedelta_series, TimedeltaSeries), pd.Series, pd.Timedelta
+    )
+    check(assert_type(td + as_period_index, pd.PeriodIndex), pd.PeriodIndex)
+    check(assert_type(td + as_datetime_index, pd.DatetimeIndex), pd.DatetimeIndex)
+    check(
+        assert_type(td + as_ndarray_td64, npt.NDArray[np.timedelta64]),
+        np.ndarray,
+        np.timedelta64,
+    )
+    check(
+        assert_type(td + as_ndarray_dt64, npt.NDArray[np.datetime64]),
+        np.ndarray,
+        np.datetime64,
+    )
+    check(assert_type(td + as_nat, NaTType), NaTType)
+
+    check(assert_type(as_period + td, pd.Period), pd.Period)
+    check(assert_type(as_timestamp + td, pd.Timestamp), pd.Timestamp)
+    check(assert_type(as_datetime + td, dt.datetime), dt.datetime)
+    check(assert_type(as_date + td, dt.date), dt.date)
+    check(assert_type(as_datetime64 + td, pd.Timestamp), pd.Timestamp)
+    # pyright can't know that as6 + td calls td.__radd__(as6),  not timedelta.__sub__
```

Comment:
fix the comment here.  should say "pyright can't know that as_td_timedelta+ td calls td.__radd__(as_td_timedelta), not timedelta.__add__"

Reply:
Fixed.